cdap-security/src/main/java/io/cdap/cdap/security/authorization/AuthorizationUtil.java

```diff
@@ -74,17 +75,27 @@ public static void ensureOnePrivilege(io.cdap.cdap.proto.id.EntityId entityId, S
                                         AuthorizationEnforcer authorizationEnforcer,
                                         Principal principal) throws Exception {
     boolean isAuthorized = false;
+    boolean includePrincipal = true;
+    String entityString = entityId.toString();
+    String addendum = null;
+    Set<String> missingPermissions = new LinkedHashSet<>();
     for (Action action : actionSet) {
       try {
         authorizationEnforcer.enforce(entityId, principal, action);
         isAuthorized = true;
         break;
       } catch (UnauthorizedException e) {
-        // continue to next action
+        missingPermissions.addAll(e.getMissingPermissions());
+        if (!e.includePrincipal()) {
```

Comment:
nit: includePrinciple = e.includePrincipal()

Reply:
Prefer the original method, since this will use the last exception thrown instead of using it if any exception has that option.